### PR TITLE
Clustered RabbitMQ

### LINKED ISSUE #111: RabbitMQ Cluster/Replicas
There is currently only a single rabbitmq sts deployed for each coinstack which will result in down time whenever kube nodes roll. It would be best if we could update to use a rabbitmq cluster or replica set deployment, similar to mongodb

### PR DESCRIPTION
**Pull request by phewitt**
Adds a RabbitMQ Cluster Operator custom resource definition so that our coinstacks can leverage a highly available RabbitMQ cluster.

### CODE REVIEW

File: coinstacks/bitcoin/docker-compose.yml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 version: '3.6'
 
+x-env: &env
```
Comment (kaladinlight) [inline coinstacks/bitcoin/docker-compose.yml:3]: Cool!

File: coinstacks/bitcoin/pulumi/Pulumi.sample.yaml [hunk 2]
```diff
@@ -33,6 +33,13 @@ config:
       replicaCount: 1
       storageSize: '1Gi'
 
+    ## rabbit - if specified a rabbit cluster will be deployed
+    rabbit:
+      cpuLimit: '500m'
+      memoryLimit: '512Mi'
+      replicaCount: 1
+      storageSize: '5Gi'
```
Comment (kaladinlight) [inline coinstacks/bitcoin/pulumi/Pulumi.sample.yaml:41]: Sample defaults are meant to be minimal case for local kube dev. Is `5Gi` required by cluster or can we go with `1Gi`?

File: coinstacks/common/pulumi/src/rabbitmq.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+export async function deployRabbit(
+  app: string,
+  asset: string,
+  provider: k8s.Provider,
+  namespace: string,
+  config: Pick<Config, 'rabbit'>
+  ): Promise<CustomResource | undefined> {
+
```
Comment (kaladinlight) [inline coinstacks/common/pulumi/src/rabbitmq.ts:22]: There is typically a check `if (config.rabbit === undefined) return` to not deploy the rabbit cluster if not defined depending on config requirement.

File: coinstacks/common/pulumi/src/rabbitmq.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  const clusterName = `${asset}-rabbitmq`
+
+  const resources: pulumi.Input<types.input.rabbitmq.v1beta1.RabbitmqClusterSpecResourcesArgs> = {
+    limits: {
+      cpu: config.rabbit?.cpuLimit as pulumi.Input<string>,
```
Comment (kaladinlight) [inline coinstacks/common/pulumi/src/rabbitmq.ts:27]: Can strings not be fed into an input?

**Review by kaladinlight — APPROVED** [review]

File: coinstacks/common/pulumi/src/rabbitmq.ts (hunk 4 above)
Comment (samortaylor) [reply coinstacks/common/pulumi/src/rabbitmq.ts:27]: I think there might be a cleaner way to specify these values as `pulumi.Input`s, they cannot be strings.

**Pull request merged**